lib/pathMatch.js

```diff
@@ -220,6 +219,14 @@ pathMatch.matchContainsJSON = function(check) {
 
 // performs a complete JSON type match with Joi
 pathMatch.matchJSONTypes = function(check) {
+  // Since Joi is an optional depenency, only attempt to load it if it's used.
+  let Joi
+  try {
+    Joi = require('joi')
+  } catch(e) {
+    throw Error('Joi is required to use expectJSONTypes, and must be installed separately')
```

Comment:
Generally I use `throw Error` instead of `throw new Error` because the they are equivalent according to the spec.

[In ES2015](https://www.ecma-international.org/ecma-262/6.0/#sec-error-constructor):

> When Error is called as a function rather than as a constructor, it creates and initializes a new Error object. Thus the function call Error(…) is equivalent to the object creation expression new Error(…) with the same arguments.

This was also true [in ES5.](http://es5.github.io/#x15.11.1)

Reply:
I'm learning a ton of things collaborating with you on this! Thanks!

Reply:
Me too! It's like testing calisthenics over here. Thanks! 💪😄